src/main/java/com/faforever/client/vault/search/SearchController.java

```diff
@@ -177,18 +177,24 @@ private void addInvalidationListener(LogicalNodeController logicalNodeController
     logicalNodeController.specificationController.propertyField.valueProperty().addListener(queryInvalidationListener);
     logicalNodeController.specificationController.operationField.valueProperty().addListener(queryInvalidationListener);
     logicalNodeController.specificationController.valueField.valueProperty().addListener(queryInvalidationListener);
-    logicalNodeController.specificationController.valueField.getEditor().textProperty().addListener(observable -> {
-      if (!logicalNodeController.specificationController.valueField.valueProperty().isBound()) {
-        JavaFxUtil.runLater(() -> logicalNodeController.specificationController.valueField.setValue(logicalNodeController.specificationController.valueField.getEditor().getText()));
-      }
-    });
+    selectValueBeforeLoosingFocusOnValueField(logicalNodeController);
     logicalNodeController.specificationController.valueField.setOnKeyReleased(event -> {
       if (event.getCode() == KeyCode.ENTER) {
         searchButton.fire();
       }
     });
   }
 
+  private void selectValueBeforeLoosingFocusOnValueField(LogicalNodeController logicalNodeController) {
```

Comment:
Why we need this?
Well if you start typing something in a free text field, then you want to see it life in the search query string. Default behaviour would be only to take effect if the combo box looses focus.
What's the issue?
This hack only works on free text fields or if the items list contains raw strings as items. In case the list is say booleans, the it calls setValue("false") instead of setValue(false). This will cause the combo box to deselect. This will also trigger whenever the user tries to select something. Anyways if there is something in the list to select this feature is not that interesting. It's only needed for free text fields anyways so I now avoid it triggering on none free text fields.